src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -737,11 +737,12 @@ abstract class CreateOp extends OpCode(0xf0, 3, 1, _.G_create) {
       case Some(err) =>
         val world2 = if (err == InvalidCall) state.world else world1
         val stack2 = stack1.push(UInt256.Zero)
+        val returnData = if (err == RevertOccurs) result.returnData else ByteString.empty
```

Comment:
It is ok, but I am not sure if it is a problem. `def withError(error: ProgramError): ProgramState[W, S] =
    copy(error = Some(error), returnData = ByteString.empty, halted = true)` - Method withError always set a buffer to empty and method `revert` set proper buffer.

Reply:
in https://github.com/input-output-hk/mantis/pull/553/files#diff-5b7d8817413ce869c362b5398e9a65f1L129 we are not using `withError` and i am not sure we should

Reply:
In my opinion, `withError` is the more informative way to do the same thing (`OutOfGas` is an error so why not?). We wouldn't need to think about checking error type, etc. But if you think your approach is clear enough I am ok with that.